Regression in v1.0.0-beta.7

Change https://github.com/deltachat/deltachat-core-rust/pull/5850 caused problems compiling https://github.com/n0-computer/iroh/

To reproduce, clone https://github.com/n0-computer/iroh/, checkout commit 29d2e82a577ebc8cb4029c0df0138fe662031d5c, go into `iroh-net` folder, run `cargo update -p derive_more` (it is pinned to v1.0.0-beta.6 in Cargo.lock) and run `cargo check`.

Compilation fails with this error:
```
$ cargo check
    Checking iroh-net v0.21.0 (/home/user/src/iroh/iroh-net)
error[E0277]: `iroh_quinn::Accept<'_>` doesn't implement `derive_more::Debug`
   --> iroh-net/src/endpoint.rs:897:10
    |
897 | #[derive(Debug)]
    |          ^^^^^ `iroh_quinn::Accept<'_>` cannot be formatted using `{:?}` because it doesn't implement `derive_more::Debug`
    |
    = help: the trait `derive_more::Debug` is not implemented for `iroh_quinn::Accept<'_>`, which is required by `&iroh_quinn::Accept<'_>: derive_more::Debug`
    = note: required for `&iroh_quinn::Accept<'_>` to implement `derive_more::Debug`
    = note: required for the cast from `&&iroh_quinn::Accept<'_>` to `&dyn derive_more::Debug`
    = note: this error originates in the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `iroh-net` (lib) due to 1 previous error
```

iroh devs worked around the problem in PR https://github.com/n0-computer/iroh/pull/2578, but this does not look like an actual fix.

--- a/tests/compile_fail/debug/lifetime_no_debug.rs
+++ b/tests/compile_fail/debug/lifetime_no_debug.rs
@@ -1,0 +1,10 @@
+struct NoDebug<'a> {
+    a: &'a f64,
+}
+
+#[derive(derive_more::Debug)]
+struct SomeType<'a> {
+    no_debug: NoDebug<'a>,
+}
+
+fn main() {}

--- a/tests/compile_fail/debug/lifetime_no_debug.stderr
+++ b/tests/compile_fail/debug/lifetime_no_debug.stderr
@@ -1,0 +1,16 @@
+error[E0277]: `NoDebug<'_>` doesn't implement `Debug`
+ --> tests/compile_fail/debug/lifetime_no_debug.rs:5:10
+  |
+5 | #[derive(derive_more::Debug)]
+  |          ^^^^^^^^^^^^^^^^^^ `NoDebug<'_>` cannot be formatted using `{:?}`
+  |
+  = help: the trait `Debug` is not implemented for `NoDebug<'_>`, which is required by `&NoDebug<'_>: Debug`
+  = note: add `#[derive(Debug)]` to `NoDebug<'_>` or manually `impl Debug for NoDebug<'_>`
+  = note: required for `&NoDebug<'_>` to implement `Debug`
+  = note: required for the cast from `&&NoDebug<'_>` to `&dyn Debug`
+  = note: this error originates in the derive macro `derive_more::Debug` (in Nightly builds, run with -Z macro-backtrace for more info)
+help: consider annotating `NoDebug<'_>` with `#[derive(Debug)]`
+  |
+1 + #[derive(Debug)]
+2 | struct NoDebug<'a> {
+  |

